### PR TITLE
use static QCoreApplication::processEvents() function without a QApplication instance.

### PR DESCRIPTION
**Pull request by ygerlach**
This removes the need for the `setApp()` call.

### Description

As seen in the Qt [documentation](https://doc.qt.io/qt-5/qcoreapplication.html#processEvents) `processEvents` is a static function so no member of `QCoreApplication` is required. Removing this takes away the need for `setApp()` in `VisualizationFrame` and `VisualizerApp` which i have keept for now, but should be marked as deprecated and get removed in the future.
Removing this requirement of QApplication help to ease the step of integrating rviz into other possibly quite big Qt-Applications, if there is no need to pass around a pointer to the main Application.

### Checklist

- [ ] If you are addressing rendering issues, please provide:
  - [ ] Images of both, broken and fixed renderings.
  - [ ] Source code to reproduce the issue, e.g. a `YAML` or `rosbag` file with a `MarkerArray` msg.
- [ ] If you are changing GUI, please include screenshots showing how things looked *before* and *after*.
- [ ] Choose the proper target branch: latest release branch, for non-ABI-breaking changes, *future* release branch otherwise.
      Due to the lack of active maintainers, we cannot provide support for older release branches anymore.
- [ ] Did you change how RViz works? Added new functionality? Do not forget to update the tutorials and/or documentation on the [ROS wiki](http://wiki.ros.org/rviz)
- [ ] While waiting for someone to review your request, please consider reviewing [another open pull request](https://github.com/ros-visualization/rviz/pulls) to support the maintainers of RViz. Refer to the [RViz Wiki](https://github.com/ros-visualization/rviz/wiki/Maintainer-Guide#reviewing-pull-requests) for reviewing guidelines.

### CODE REVIEW

File: src/rviz/visualization_frame.h [hunk 1]
```diff
@@ -78,6 +78,7 @@ class RVIZ_EXPORT VisualizationFrame : public QMainWindow, public WindowManagerI
   VisualizationFrame(QWidget* parent = nullptr);
   ~VisualizationFrame() override;
 
+  // now deprecated
```
Comment (rhaschke) [inline src/rviz/visualization_frame.h:81]:
```suggestion
  [[deprecated("setApp() not needed anymore")]]
```

File: src/rviz/visualizer_app.h [hunk 2]
```diff
@@ -51,6 +51,7 @@ class RVIZ_EXPORT VisualizerApp : public QObject
   VisualizerApp();
   ~VisualizerApp() override;
 
+  // now deprecated
```
Comment (rhaschke) [inline src/rviz/visualizer_app.h:54]:
```suggestion
  [[deprecated("setApp() not needed anymore")]]
```

**Review by rhaschke — COMMENTED** [review]
Looks good. A small nitpick though:

**Comment by ygerlach** [comment]
good, i wasn't sure how deprecated methods are annotated in this project.

**Comment by ygerlach** [comment]
What needs to happen, that this MR can be merged?

**Comment by rhaschke** [comment]
Sorry, this got out of my sight. Closing and reopening to trigger CI again...

**Pull request merged**